docs/USAGE-GUIDE.md

```diff
@@ -343,6 +343,33 @@ An application can override s2n-tls’s internal memory management by calling `s
 
 If you are trying to use FIPS mode, you must enable FIPS in your libcrypto library (probably by calling `FIPS_mode_set(1)`) before calling `s2n_init()`.
 
+## Connection
+
+Users will need to create a `s2n_connection` struct to store all of the state necessary to negotiate a TLS connection. Call `s2n_connection_new()` to create a new server or client connection. Call `s2n_connection_free()` to free the memory allocated for this struct when no longer needed.
+
+### Connection Memory
+
+The connection struct is roughly 4KB with some variation depending on how it is configured. Maintainers of the s2n-tls library carefully consider increases to the size of the connection struct as they are aware some users are memory-constrained.
+
+The TLS handshake requires a significant amount of memory, but if necessary the handshake memory can be freed by calling `s2n_connection_free_handshake()`. This function should be called after the handshake is successfully negotiated and logging or recording of handshake data is complete. This function should not be called if you're planning to reuse the connection object for multiple connections. See [connection reuse](#connection-reuse).
+
```

Comment:
I'm still not sure this is quite right. Using s2n_connection_wipe is the most common reason you wouldn't call s2n_connection_free_handshake, but you can still call it if you're planning to reuse the connection-- maybe connections are pooled but also very long-lived?

Maybe:
> An s2n-tls connection allocates memory for the TLS handshake that isn't required when later sending and receiving application data. This memory can be freed by calling `s2n_connection_free_handshake`, but some functions that print information about the handshake may not produce meaningful results after the handshake memory is freed. Applications may want to free handshake memory due to memory constraints, particularly if connections are long-lived. However, the handshake memory can be reused for another connection if `s2n_connection_wipe` is called, so freeing it may later result in more potentially expensive memory allocations.

Reply:
Didn't copy this paragraph word for word, but I did pull a lot from it.